### PR TITLE
added inorder.cpp

### PR DESCRIPTION
**Pull request by shivani1706**
#### Description of Change

added a new file inorder.cpp which contains the code for algorithmic implementation of inorder traversal of trees using cpp

#### Checklist

- [x ] Added description of change
- [x ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [ x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x ] Relevant documentation/comments is changed or added
- [ x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x ] I acknowledge that all my contributions will be made under the project's license.

Notes: 
added implementation of inorder traversal of trees in cpp

solved issue "#2014"

### CODE REVIEW

File: data_structures/inorder.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+{
+    int maze[N][N] = {{1, 0, 0, 0},
+                      {1, 1, 0, 1},
+                      {0, 1, 0, 0},
+                      {1, 1, 1, 1}};
+    solveMaze(maze);
+    return 0;
+}
+
+#endif // RAT_in_A_MAZE_HPP
```
Comment (mishraabhinn) [inline data_structures/inorder.cpp:114]: Please read contributing guidelines. 
I feel copied from https://www.geeksforgeeks.org/rat-in-a-maze-backtracking-2/
Please avoid plagiarism.

**Review by mishraabhinn — CHANGES_REQUESTED** [review]

**Comment by manishtambe** [comment]
@mishraabhinn if this issue is still open can you please assign it to me ?

**Comment by thewpingmonk** [comment]
hello sir can you assign this issue to me?

**Comment by VishalVR-21** [comment]
Please assign me this

**Pull request closed without merging**